rules/linux/credential_access_unusual_instance_metadata_service_api_request.toml

```diff
@@ -0,0 +1,83 @@
+[metadata]
+creation_date = "2024/08/22"
+integration = ["endpoint"]
+maturity = "production"
+updated_date = "2024/08/22"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies potentially malicious processes attempting to access the cloud service provider's instance metadata service (IMDS) API
+endpoint, which is typically used to retrieve sensitive instance-specific information such as instance ID, public IP
+address, and even temporary security credentials if role's are assumed by that instance. The rule monitors for various tools and scripts like curl, wget,
+python, and perl that might be used to interact with the metadata API.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Unusual Instance Metadata Service (IMDS) API Request"
+references = ["https://hackingthe.cloud/aws/general-knowledge/intro_metadata_service/"]
+risk_score = 47
+rule_id = "ecc0cd54-608e-11ef-ab6d-f661ea17fbce"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: Linux",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Tactic: Discovery",
+    "Data Source: Elastic Defend",
+]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+host.os.type: "linux" and event.category:process and event.type:start and event.action:exec and
+(
+    (
+        (
+            process.name:(
+                curl or wget or python* or perl* or php* or ruby* or lua* or telnet or pwsh or
+                openssl or nc or ncat or netcat or awk or gawk or mawk or nawk or socat or node
+            ) or
+            process.executable: (
+                ./* or /tmp/* or /var/tmp/* or /var/www/* or /dev/shm/* or /etc/init.d/* or
+                /etc/rc*.d/* or /etc/cron* or /etc/update-motd.d/* or /boot/* or /srv/* or
+                /run/* or /etc/rc.local
+            )
+        ) and destination.ip: (169.254.169.254 or "fd00:ec2::254")
```

Comment:
`destination.ip` can't be used with the `process` category. So as we discussed on Slack, you could either choose to be category-less and make fields such as `process.working_directory` and `process.parent.executable` optional, or use a `sequence` with a process --> network event, which allows "best of both worlds".

Reply:
One other approach could be to just monitor for network events only, and white list processes + executables. Only risk this approach would have is that it is difficult to tune, given that Linux network events do not have parent fields populated.